### PR TITLE
Chore: Update Makefile to support go workspace

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

Adds support for the multi-module workspace in `make test-go` and make lint-go` targets.

**Why do we need this feature?**

Sub packages are skipped when using `...`. See https://github.com/golang/go/issues/50745 for more info.

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -215,11 +215,11 @@ test: test-go test-js ## Run all tests.
 ##@ Linting
 golangci-lint: $(GOLANGCI_LINT)
 	@echo "lint via golangci-lint"
+	go list -f '{{.Dir}}/...' -m | xargs \
 	$(GOLANGCI_LINT) run \
-		--config .golangci.toml \
-		$(GO_FILES)
```
Comment (toddtreece) [inline Makefile:220]: it doesn't seem like `$(GO_FILES)` is currently used anywhere

File: scripts/drone/events/pr.star [hunk 2]
```diff
@@ -96,6 +96,7 @@ def pr_pipelines():
         test_backend(
             get_pr_trigger(
                 include_paths = [
+                    "Makefile",
```
Comment (toddtreece) [inline scripts/drone/events/pr.star:99]: changing the makefile didn't trigger go tests

**Review by charandas — APPROVED** [review]

File: scripts/drone/events/pr.star (hunk 2 above)
Comment (charandas) [reply scripts/drone/events/pr.star:99]: Oh I see, you already addressed it.

Comment (toddtreece) [reply scripts/drone/events/pr.star:99]: yeah, sorry, the way i worded that wasn't clear

**Pull request merged**